### PR TITLE
Feature/SK-1088 | Added mnist-libtorch example

### PR DESCRIPTION
**Pull request by benjaminastrand**
Added example _mnist-libtorch_ - corresponds to the _mnist-pytorch_ example in the fedn repo.
* Uses the same neural network as in mnist-pytorch (struct Net).
* The train() function loads the parameters from the bin file, trains the model and writes the params to file again in binary format.
* ML code uses libtorch - PyTorch's C++ API. Note that the train and test datasets are downloaded from scaleout public bucket.
* This example showcases the differences and similarities between implementing the ML code in a python _compute package_ and by overriding functions in the C++ class GrpcClient.
* In this example, the entire MNIST dataset is used for each client - no partitioning is implemented.
* Also created a unique directory and CMake file for each example to keep things organized.
* **To start:** Start the client using a downloaded client.yaml file. Create a session with the seed.bin that is located in the mnist-libtorch directory.

### CODE REVIEW

**Comment by benjaminastrand** [comment]
When starting the client from terminal, MNIST dataset is downloaded from GCP, if it is doesn't already exist locally.

**Review by Wrede — APPROVED** [review]
